core/src/main/scala/gnieh/diffson/JsonPointerSupport.scala

```diff
@@ -18,34 +18,67 @@ package gnieh.diffson
 
 import scala.annotation.tailrec
 
+import scala.language.implicitConversions
+
 trait JsonPointerSupport[JsValue] {
   this: DiffsonInstance[JsValue] =>
 
   import provider._
 
-  implicit val pointer = new JsonPointer()
-
   /** A class to work with Json pointers according to http://tools.ietf.org/html/rfc6901.
    *  The behavior in case of invalid pointer is customizable by passing an error handler
    *  when instantiating.
    *
    *  @author Lucas Satabin
    */
-  class JsonPointer {
+  case class JsonPointer(path: Pointer) {
+
+    /** Evaluates the given path in the given JSON object.
+     *  Upon missing elements in value, the error handler is called with the current value and element
+     */
+    final def evaluate(value: JsValue): JsValue = evaluate(value, this, Pointer.Root)
```

Comment:
There is already `defaultHandler` in scope, which can be overridden in class extending `JsonProvider`. If I add it here, I am getting error message:
```
[error]  both method errorHandler in class JsonProvider of type => JsonPointerSupport.this.provider.PointerErrorHandler
[error]  and value handler of type JsonPointerSupport.this.provider.PointerErrorHandler
[error]  match expected type JsonPointerSupport.this.provider.PointerErrorHandler
[error]       evaluate(value, this, Pointer.Root)
```
Exactly as expected... I have to either put it here as explicit argument, or remove `errorHandler` from JsonProvider.